core/src/smartcontracts/isi/asset.rs

```diff
@@ -335,7 +334,89 @@ pub mod isi {
                     .ok_or(MathError::Overflow)?;
                 wsv.metrics.tx_amounts.observe(f64::from(*quantity));
 
-                Ok(AssetEvent::Added(destination_asset_id.clone()))
+                Ok(AssetEvent::Added(self.destination_id.clone()))
+            })
+        }
+    }
+
+    impl<W: WorldTrait> Execute<W> for Transfer<Asset, u128, Asset> {
+        type Error = Error;
+
+        #[metrics(+"transfer_qty_asset")]
```

Comment:
Shouldn't this be changed? A copy-paste mistake, I believe

Reply:
Yes.